pkg/reconciler/bitbucketsource/webhook_client.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bitbucketsource
+
+import (
+	"context"
+	"fmt"
+	bbclient "github.com/knative/eventing-sources/pkg/reconciler/bitbucketsource/resources"
+	"github.com/knative/pkg/logging"
+	"golang.org/x/oauth2/bitbucket"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type webhookOptions struct {
```

Comment:
I am guessing because of this package's location, there is no need to export any of these but it looks suspect. I would expect all the options to be exported.

Reply:
This is just a helper struct to reduce the number of arguments in a couple of functions. It's being used from the controller, so it's fine to have everything unexported. 
The one that exports stuff is Hook in the actual client code that I moved outside the package.
I kind of follow github source example. I think this should be fine.